src/components/Product.js

```diff
@@ -0,0 +1,175 @@
+import { useEffect, useState } from 'react';
+import { FontAwesomeIcon } from "@fortawesome/react-fontawesome";
+import { faDollarSign } from '@fortawesome/fontawesome-free-solid';
+import { Carousel } from "react-responsive-carousel";
+import "react-responsive-carousel/lib/styles/carousel.min.css";
+import { ToastContainer, toast } from 'react-toastify';
+import 'react-toastify/dist/ReactToastify.css';
+
+function Product({cart, setCart, productInfo}) {
+
+    const [product, setProduct] = useState({});
+    const [loaded, setLoaded] = useState(false);
+    const [selectedSize, setSelectedSize] = useState('');
+    const [selectedQuantity, setSelectedQuantity] = useState(1);
+
+    const fetchProductData = (product_id) => {
+      fetch("http://3.86.87.11:8080/getProductDetails/" + product_id)
+        .then(response => {
+          return response.json()
+        })
+        .then(data => {
+          setProduct(data[0]);
+          setLoaded(true);
+        })
+    }
+
+    function getDiscount(newPrice, oldPrice){
+      var diff = oldPrice - newPrice;
+      return Math.round(diff*100/oldPrice);
+    }
+    
+    useEffect(() => {
+      fetchProductData(productInfo.product_id);
+    },[productInfo.product_id]);
+
+    function addItemToCartHandler(){
+      
+      var productSku = product.product_id.toString() + '#' + selectedSize.toString();
+      var newCartItem = {};
+      if(cart[productSku]){
+        newCartItem = {...cart[productSku], quantity: cart[productSku].quantity + parseInt(selectedQuantity)};
+      } else {
+        newCartItem = {size:selectedSize, quantity:parseInt(selectedQuantity), image:product.images.split('|')[0], title:product.title, price:product.variant_price};
```

Comment:
@rahulmeghwal not sure if this would be a problem, but I think browsers have URL length limits. Not sure what the ultimate query string length here is, but any issues there you can think of?

Reply:
The cart doesn't use any query params here, but the fetch call to get the desired products uses path params. The limit is 2048 characters for  URLs, for our use case right now I don't see our URL ( including path params) reaching near this limit. If at all such need arise we can use query params.